src/modules/sensors/sensors.cpp

```diff
@@ -394,49 +409,72 @@ void Sensors::diff_pres_poll()
 			}
 		}
 
-		airspeed_s airspeed{};
-		airspeed.timestamp = diff_pres.timestamp;
+		// push raw data into validator
+		float airspeed_input[3] { diff_pres.differential_pressure_pa, air_temperature_celsius, 0.0f };
+		_airspeed_validator.put(diff_pres.timestamp_sample, airspeed_input, diff_pres.error_count, 100); // TODO: real priority?
 
-		/* push data into validator */
-		float airspeed_input[3] = { diff_pres.differential_pressure_raw_pa, diff_pres.temperature, 0.0f };
+		// accumulate average for publication
+		_diff_pres_timestamp_sum += diff_pres.timestamp_sample;
+		_diff_pres_pressure_sum += diff_pres.differential_pressure_pa;
+		_diff_pres_temperature_sum += air_temperature_celsius;
+		_baro_pressure_sum += air_data.baro_pressure_pa;
+		_diff_pres_count++;
 
-		_airspeed_validator.put(airspeed.timestamp, airspeed_input, diff_pres.error_count, 100); // TODO: real priority?
+		if ((_diff_pres_count > 0) && hrt_elapsed_time(&_airspeed_last_publish) >= 100_ms) {
 
-		airspeed.confidence = _airspeed_validator.confidence(hrt_absolute_time());
+			// average data and apply calibration offset (SENS_DPRES_OFF)
+			const uint64_t timestamp_sample = _diff_pres_timestamp_sum / _diff_pres_count;
+			const float differential_pressure_pa = diff_pres.differential_pressure_pa / _diff_pres_count -
```

Comment:
Here we should have `_diff_pres_pressure_sum` instead of `diff_pres.differential_pressure_pa`.
This mistake leads to the scale I mentioned above.

Reply:
Ouch, thanks @PonomarevDA.